src/test/main.test.js

```diff
@@ -376,41 +341,6 @@ describe('api', function() {
     });
 
 
-    it('public key validation', async () => {
```

Comment:
Why was this removed? This seems especially important to the change.

Reply:
refactored into 'rejects invalid key' and 'rejects non-matching keys'